src/colorizer/loaders/JsonArrayLoader.ts

```diff
@@ -57,13 +60,37 @@ class JsonArraySource implements ArraySource {
   }
 }
 
-const nanToNull = (json: string): string => json.replace(/NaN/g, "null");
-
-export default class JsonArrayLoader implements IArrayLoader {
-  async load(url: string): Promise<JsonArraySource> {
-    const response = await fetch(url);
-    const text = await response.text();
-    const { data, min, max }: FeatureDataJson = JSON.parse(nanToNull(text));
-    return new JsonArraySource(data, min, max);
+export default class UrlArrayLoader implements IArrayLoader {
+  /**
+   * Loads array data from the specified URL, handling both JSON and Parquet files.
+   * @param url The URL to load data from. Must end in ".json" or ".parquet".
+   * @param min Optional minimum value for the data. If defined, overrides the `min` value
+   *   in the JSON file.
+   * @param max Optional maximum value for the data. If defined, overrides the `max` value
+   *   in the JSON file.
+   * @throws Error if the file format is not supported (not JSON or Parquet).
+   * @returns a URLArraySource object containing the loaded data.
+   */
+  async load(url: string, min?: number, max?: number): Promise<UrlArraySource> {
+    if (url.endsWith(".json")) {
+      const response = await fetch(url);
+      const text = await response.text();
+      const { data, min: jsonMin, max: jsonMax }: FeatureDataJson = JSON.parse(nanToNull(text));
+      return new UrlArraySource(data, min ?? jsonMin, max ?? jsonMax);
+    } else if (url.endsWith(".parquet")) {
+      const result = await fetch(url);
+      const arrayBuffer = await result.arrayBuffer();
+      let data: number[] = [];
+      await parquetRead({
+        file: arrayBuffer,
+        compressors,
+        onComplete: (loadedData: number[][]) => {
+          data = loadedData.map((row) => Number(row[0]));
```

Comment:
1. Since the array may be large, it might be worth benchmarking `map` against a traditional `for` loop or iterator (`for row of loadedData) { ... }`
2. If you have to loop through the data anyways, could you also be accumulating into a `min` and `max` now?
3. In fact, is there a way the `min` and `max` can come along with the data, as it can in json?

Reply:
1. Ooh, good point, I'll do that.
2. That's a good idea in case the user didn't provide a min and max, though `colorizer-data` should guarantee that parquet data has one. `min` and `max` are meant to be overridden by the user, though, for example in the case where outliers would blow the min/max out of proportion.
3. It might be possible if I include a "min" and "max" column, but if we ever pursue async loading again it's actually much better to have the feature min/max live inside the manifest JSON. That way we have immediate access to the range even if the feature data hasn't been loaded in.